### DISCUSSION TITLE
will DRI_PRIME work with xrdp (Xorg) and dgpu?

### DISCUSSION THREAD

**Discussion by zeusin** [neutrinolabs/xrdp#2373] [Q&A]
Hello guys,

I'm testing xrdp and came to my mind if using the offloading capacities of xorg will help xrdp with xorg server to have 3d hardware acceleration on applications running with DRI_PRIME=1

Can't find any related topic and I don't have an external gpu to try now.

**Comment by matt335672** [comment]
Hi @zeusin 

This is somewhat out of my area of expertise I'm afraid. I'll move it the the Q&A section where it will hopefully get more visibility.

**Comment by akarl10** [comment]
as far as I know in Xorg you can add additional devices as GPUDevice, but in practice the drivers of that GPUs pull in the "modesetting" X driver. This one is unable to load in the xorgxrdp case and stops the entire X instance.

You should get 3d hardware acceleration if you enable glamor in xorgxrdp (and additionally you need rw permissions on /dev/dri/renderD128), but of course this is single gpu only
